src/datamodel/game/player.rs

```diff
@@ -0,0 +1,6 @@
+use crate::datamodel::character::Character;
+
+pub struct Player {
+    pub name: String,
+    pub character: Box<dyn Character>,
```

Comment:
What's the use for box here? just for my personal knowledge, I don't see anything wrong with it.

Reply:
My understanding is that we can not create a `Vec<Character>` in the stack because the compiler does not know how much memory to allocate, since `Character` is a trait and different implementations would consume different amounts of memory (I think that's what the `dyn` indicates). Wrapping it into a [`Box`](https://doc.rust-lang.org/std/boxed/struct.Box.html) makes it possible for the `Player` class to implement the [`Sized`](https://doc.rust-lang.org/std/marker/trait.Sized.html) trait since the space occupied in the stack is that of the box (see [the book](https://doc.rust-lang.org/book/ch15-01-box.html#using-boxt-to-get-a-recursive-type-with-a-known-size)), and therefor be allocated in the stack in in a `Vec` (eg. the `Game` class has a `Vec<Player>` field).

Reply:
Nice input @julianmateu